### PR TITLE
feat (coupons): add changes related to the new coupon fields

### PR DESCRIPTION
**Pull request by lovrocolic**

### CODE REVIEW

File: docs/api/07_coupons/apply-coupon.mdx [hunk 1]
```diff
@@ -136,6 +149,9 @@ import TabItem from '@theme/TabItem';
...
 | amount_cents | Integer &nbsp &nbsp &nbsp<Optional>**Optional**</Optional> | Amount to apply to the customer.<br></br>If defined, it overrides the amount_cents property of the coupon |
 | amount_currency | String &nbsp &nbsp &nbsp<Optional>**Optional**</Optional> | Currency of the amount to apply to the customer.<br></br>If defined, it overrides the amount_currency property of the coupon.<br></br>It must match the currency of the customer's plan. |
+| percentage_rate | Float &nbsp &nbsp &nbsp<Optional>**Optional**</Optional> | Rate to apply with the coupon.<br></br>If defined, it overrides the percentage_rate property of the coupon |
+| frequency | String &nbsp &nbsp &nbsp<Optional>**Optional**</Optional> | Frequency type.<br></br>It can be `once` or `recurring`.<br></br>If defined, it overrides the frequency property of the coupon |
+| frequency_duration | Integer &nbsp &nbsp &nbsp<Optional>**Optional**</Optional> | This field determines number of billing periods to which the coupon is applied.<br></br>If defined, it overrides the frequency_duration property of the coupon |
```
Comment (mathieu-lago) [inline docs/api/07_coupons/apply-coupon.mdx:154]: This field determines the number of billing periods to which the coupon applies.

File: docs/api/07_coupons/apply-coupon.mdx [hunk 2]
```diff
@@ -251,6 +270,9 @@ For now, it lets you assign a one-off coupon to a customer. This coupon can have
 | **lago_customer_id** &nbsp &nbsp <Type>String</Type> &nbsp &nbsp <NotNullable>Not null</NotNullable> | Unique identifer of the customer in Lago application. |
 | **amount_cents** &nbsp &nbsp <Type>Integer</Type> &nbsp &nbsp <NotNullable>Not null</NotNullable> | Amount in cents. |
 | **amount_currency** &nbsp &nbsp <Type>String</Type> &nbsp &nbsp <NotNullable>Not null</NotNullable> | Currency of the amount. |
+| **percentage_rate** &nbsp &nbsp <Type>Float</Type> | Percentage rate of the coupon. |
+| **frequency** &nbsp &nbsp <Type>String</Type> &nbsp &nbsp <NotNullable>Not null</NotNullable><br></br> | Frequency type. <br></br> <details><summary>**Possible values**</summary><div>- `once`<br></br>- `recurring`<div></div></div></details> |
+| **frequency_duration** &nbsp &nbsp <Type>Integer</Type> | It determines number of billing periods to which the coupon is applied |
```
Comment (mathieu-lago) [inline docs/api/07_coupons/apply-coupon.mdx:275]: It determines the number of billing periods to which the coupon applies.

File: docs/api/07_coupons/coupon-object.mdx [hunk 3]
```diff
@@ -28,10 +32,14 @@ This object represents a coupon.<br></br>
...
 | **name** &nbsp &nbsp <Type>String</Type> &nbsp &nbsp <NotNullable>Not null</NotNullable> | Coupon name. |
 | **code** &nbsp &nbsp <Type>String</Type> &nbsp &nbsp <NotNullable>Not null</NotNullable> | Code identifying the coupon. |
-| **amount_cents** &nbsp &nbsp <Type>Integer</Type> &nbsp &nbsp <NotNullable>Not null</NotNullable> | Amount in cents. |
-| **amount_currency** &nbsp &nbsp <Type>String</Type> &nbsp &nbsp <NotNullable>Not null</NotNullable> | Currency of the amount. |
+| **amount_cents** &nbsp &nbsp <Type>Integer</Type> | Amount in cents. |
+| **amount_currency** &nbsp &nbsp <Type>String</Type> | Currency of the amount. |
 | **expiration** &nbsp &nbsp <Type>String</Type> &nbsp &nbsp <NotNullable>Not null</NotNullable><br></br> | Expiration type. <br></br> <details><summary>**Possible values**</summary><div>- `time_limit`<br></br>- `no_expiration`<div></div></div></details> |
-| **expiration_duration** &nbsp &nbsp <Type>Integer</Type> | Duration of the coupon. |
+| **expiration_date** &nbsp &nbsp <Type>String</Type> &nbsp &nbsp <NotNullable>Not null</NotNullable> &nbsp &nbsp <br></br><Comment>*ISO 8601 datetime in UTC*</Comment> | Date of the coupon expiration. |
```
Comment (mathieu-lago) [inline docs/api/07_coupons/coupon-object.mdx:38]: Expiration date of the coupon.

File: docs/api/07_coupons/coupon-object.mdx [hunk 4]
```diff
@@ -28,10 +32,14 @@ This object represents a coupon.<br></br>
...
+| **coupon_type** &nbsp &nbsp <Type>String</Type> &nbsp &nbsp <NotNullable>Not null</NotNullable><br></br> | Coupon type. <br></br> <details><summary>**Possible values**</summary><div>- `fixed_amount`<br></br>- `percentage`<div></div></div></details> |
+| **percentage_rate** &nbsp &nbsp <Type>Float</Type> | Percentage rate of the coupon. |
+| **frequency** &nbsp &nbsp <Type>String</Type> &nbsp &nbsp <NotNullable>Not null</NotNullable><br></br> | Frequency type. <br></br> <details><summary>**Possible values**</summary><div>- `once`<br></br>- `recurring`<div></div></div></details> |
+| **frequency_duration** &nbsp &nbsp <Type>Integer</Type> | It determines number of billing periods to which the coupon is applied |
```
Comment (mathieu-lago) [inline docs/api/07_coupons/coupon-object.mdx:42]: It determines the number of billing periods to which the coupon applies.

File: docs/api/07_coupons/create-coupon.mdx [hunk 5]
```diff
@@ -144,10 +167,13 @@ import TabItem from '@theme/TabItem';
 |--|--|--|
 | name | String &nbsp &nbsp &nbsp<Required>**Required**</Required> | Coupon name |
 | code | String &nbsp &nbsp &nbsp<Required>**Required**</Required> | Code identifying the coupon |
-| amount_cents | Integer &nbsp &nbsp &nbsp<Required>**Required**</Required> | Amount of the coupon |
-| amount_currency | String &nbsp &nbsp &nbsp<Required>**Required**</Required> | Currency of the amount |
+| amount_cents | Integer &nbsp &nbsp &nbsp<Optional>**Optional (required only for `fixed_amount` type)**</Optional> | Amount of the coupon |
+| amount_currency | String &nbsp &nbsp &nbsp<Optional>**Optional (required only for `fixed_amount` type)**</Optional> | Currency of the amount |
 | expiration | String &nbsp &nbsp &nbsp<Required>**Required**</Required> | Expiration type.<br></br>It can be `time_limit` or `no_expiration` |
-| expiration_duration | Integer &nbsp &nbsp &nbsp<Required>**Required (optional only for `no_expiration` type)**</Required> | Coupon duration. |
+| expiration_date | String &nbsp &nbsp &nbsp<Optional>**Optional (required only for `time_limit` type)**</Optional> | Coupon expiration date. |
```
Comment (mathieu-lago) [inline docs/api/07_coupons/create-coupon.mdx:173]: Expiration date of the coupon.

File: docs/api/07_coupons/create-coupon.mdx [hunk 6]
```diff
@@ -144,10 +167,13 @@ import TabItem from '@theme/TabItem';
...
+| frequency | String &nbsp &nbsp &nbsp<Required>**Required**</Required> | Frequency type.<br></br>It can be `once` or `recurring` |
+| frequency_duration | String &nbsp &nbsp &nbsp<Optional>**Optional (required only for `recurring` frequency type)**</Optional> | This field determines number of billing periods to which the coupon is applied. |
```
Comment (mathieu-lago) [inline docs/api/07_coupons/create-coupon.mdx:175]: This field determines the number of billing periods to which the coupon applies.

File: docs/api/07_coupons/update-coupon.mdx [hunk 7]
```diff
@@ -117,19 +120,25 @@ import TabItem from '@theme/TabItem';
...
-| expiration | String &nbsp &nbsp &nbsp<Optional>**Optional**</Optional> | Expiration type.<br></br>It can be `time_limit` or `no_expiration`<br></br>It won't be updated if coupon is attached to any subscriptions |
-| expiration_duration | Integer &nbsp &nbsp &nbsp<Optional>**Optional**</Optional> | Coupon duration<br></br>It won't be updated if coupon is attached to any subscriptions |
+| name | String &nbsp &nbsp &nbsp<Required>**Required**</Required> | Coupon name |
+| code | String &nbsp &nbsp &nbsp<Required>**Required**</Required> | Code identifying the coupon |
+| amount_cents | Integer &nbsp &nbsp &nbsp<Optional>**Optional (required only for `fixed_amount` type)**</Optional> | Amount of the coupon |
+| amount_currency | String &nbsp &nbsp &nbsp<Optional>**Optional (required only for `fixed_amount` type)**</Optional> | Currency of the amount |
+| expiration | String &nbsp &nbsp &nbsp<Required>**Required**</Required> | Expiration type.<br></br>It can be `time_limit` or `no_expiration` |
+| expiration_date | String &nbsp &nbsp &nbsp<Optional>**Optional (required only for `time_limit` type)**</Optional> | Coupon expiration date. |
```
Comment (mathieu-lago) [inline docs/api/07_coupons/update-coupon.mdx:138]: Expiration date of the coupon.

File: docs/api/07_coupons/update-coupon.mdx [hunk 8]
```diff
@@ -117,19 +120,25 @@ import TabItem from '@theme/TabItem';
...
+| frequency | String &nbsp &nbsp &nbsp<Required>**Required**</Required> | Frequency type.<br></br>It can be `once` or `recurring` |
+| frequency_duration | String &nbsp &nbsp &nbsp<Optional>**Optional (required only for `recurring` frequency type)**</Optional> | This field determines number of billing periods to which the coupon is applied. |
```
Comment (mathieu-lago) [inline docs/api/07_coupons/update-coupon.mdx:140]: This field determines the number of billing periods to which the coupon applies.

**Review by mathieu-lago — APPROVED** [review]

File: docs/api/07_coupons/apply-coupon.mdx (hunk 1 above)
Comment (lovrocolic) [reply docs/api/07_coupons/apply-coupon.mdx]: Fixed grammar based on comments

**Pull request merged**